File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/graphics/GradientStop.java
package com.hello.uxml.tools.framework.graphics;

/**
 * Represents a gradient color, offset pair.
 *
 * @author ferhat
 *
 */
public class GradientStop {

  /** Color of gradient stop */
  private Color color;

  /** Offset of gradient */
  private double offset;

  /**
   * Constructor.
   */
  public GradientStop() {
  }

  /**
   * Constructor.
   */
  public GradientStop(Color color, double offset) {
    this.color = color;
    this.offset = offset;
  }

  /**
   * Returns color.
   */
  public Color getColor() {
    return color;
  }

  public void setColor(Color value) {
    color = value;
  }

  /**
   * Returns offset of gradient stop.
   */
  public double getOffset() {
    return offset;
  }

  public void setOffset(double value) {
    offset = value;
  }
}

File: framework/lib/gradient_stop.dart
part of uxml;

class GradientStop {
  /**
   * Gets or sets the color value.
   */
  Color color;

  /**
   * Gets or sets the stop's offset inside a gradient.
   */
  num offset;

  GradientStop(this.color, this.offset);

  /** Clones Gradient stop and color. */
  GradientStop clone() {
    return new GradientStop(new Color(color.argb), offset);
  }
}
